nucypher/blockchain/eth/clients.py

```diff
@@ -15,16 +16,109 @@
 )
 from geth.process import BaseGethProcess
 from twisted.logger import Logger
+from web3 import Web3
 
-from nucypher.config.constants import DEFAULT_CONFIG_ROOT, BASE_DIR, DEPLOY_DIR, USER_LOG_DIR
+from nucypher.config.constants import DEFAULT_CONFIG_ROOT, DEPLOY_DIR, USER_LOG_DIR
 
 NUCYPHER_CHAIN_IDS = {
     'devnet': 112358,
 }
 
 
-class NuCypherGethProcess(LoggingMixin, BaseGethProcess):
+class Web3ClientError(Exception):
+    pass
+
+
+class Web3ClientConnectionFailed(Web3ClientError):
+    pass
+
+
+class Web3ClientUnexpectedVersionString(Web3ClientError):
+    pass
+
+
+class Web3Client(object):
+
+    def __new__(self, w3: Web3, *args):
```

Comment:
What is the advantage of using `__new__` over a `classmethod` in this situation?

I can imagine this same logic in `Web3Client.from_version_string`.  I do not feel strongly either way, but alternate -constructor `@classmethods` are popular in the nucypher codebase.

Reply:
yep